src/execution_plan/ops/op_project.c

```diff
@@ -79,13 +79,17 @@ Record ProjectConsume(OpBase *opBase) {
 	int rec_idx = 0;
 	for(unsigned short i = 0; i < op->exp_count; i++) {
 		SIValue v = AR_EXP_Evaluate(op->exps[i], r);
```

Comment:
How about, just for the sake of clarity we define an alias to `SIValue` suggesting `SIValueVolatile` which will indicate the fact that an SIValue returned from an arithmetic expression evaluation is indeed volatile, with that `SIValue_Persists ` can accept only  `SIValueVolatile` and will return `SIValue`?

Reply:
I'm open to the idea of aliases like that, but `AR_EXP_Evaluate` doesn't necessarily return volatiles. Specifically in the case of string arithmetic ops, the returned SIValue is of M_SELF, and `SI_ShareValue` changes nothing in the very common case of stack (M_NONE) values.